workspaces/ballerina/ballerina-extension/src/features/ai/service/documentation/doc_generator.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { DocGenerationRequest, ProjectSource } from '@wso2/ballerina-core';
+import { getServiceDeclaration } from '../../testGenerator';
+import { getBallerinaProjectRoot } from '../../../../rpc-managers/ai-panel/rpc-manager';
+import { generateDocumentation, DocumentationGenerationRequest } from './documentation';
+import { workspace } from "vscode";
+import * as path from 'path';
+import * as fs from 'fs';
+
+// Main documentation generator function that handles all the logic
+export async function generateDocumentationForService(params: DocGenerationRequest): Promise<void> {
+    try {
+        // Get the project root
+        const projectRoot = await getBallerinaProjectRoot();
+
+        // Get the project source files
+        const projectSource = await getProjectSource(projectRoot);
+        if (!projectSource) {
+            throw new Error("The current project is not recognized as a valid Ballerina project. Please ensure you have opened a Ballerina project.");
+        }
+
+        // Find the service declaration and get OpenAPI spec
+        const { serviceDocFilePath } = await getServiceDeclaration(projectRoot, params.serviceName);
+        const openApiSpec = await getOpenAPISpecification(serviceDocFilePath);
+
+        // Create the documentation generation request
+        const docRequest: DocumentationGenerationRequest = {
+            serviceName: params.serviceName,
+            projectSource: projectSource,
+            openApiSpec: openApiSpec
+        };
+
+        // Generate the documentation with streaming
+        await generateDocumentation(docRequest);
+    } catch (error) {
+        console.error("Error during documentation generation:", error);
+        throw error;
+    }
+}
+
+// Internal helper functions (copied from testGenerator.ts since they're not exported)
+async function getProjectSource(dirPath: string): Promise<ProjectSource | null> {
+    try {
+        const ballerinaTomlPath = path.join(dirPath, "Ballerina.toml");
+
+        if (!fs.existsSync(ballerinaTomlPath)) {
+            return null;
+        }
+
+        const sourceFiles: any[] = [];
+        const ballerinaFiles = await workspace.findFiles("**/*.bal", "**/target/**");
```

Comment:
```suggestion
        const ballerinaFiles = await workspace.findFiles("**/*.bal", "**/target/**");
```

We need to avoid `generated` as well